sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+                initialListResult.NextPageLink);
+
+            var nextListedResolvers = nextListResult.ToArray();
+            nextListedResolvers.Count().Should().Be(numDnsResolvers - topValue);
+            nextListedResolvers.All(resolver => ValidateListedResolverIsExpected(resolver, expectedResolver));
+        }
+
+        [Fact]
+        public void ListDnsResolverInResourceGroup_MultipleResolversPresentWithTopZero_ExpectError()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var numDnsResolvers = 2;
+            var topValue = 0;
+            var createdDnsResolvers = this.CreateDnsResolvers(resourceGroupName: resourceGroupName, numDnsResolvers: numDnsResolvers);
+
+            Action listDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.ListByResourceGroup(
+                resourceGroupName: resourceGroupName,
+                top: topValue);
+
+            listDnsResolverAction.Should().NotBeNull();
+            listDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverNotExists_ExpectError()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolverForPatch = TestDataGenerator.GenerateDnsResolver(resourceGroupName: resourceGroupName);
+
+            Action patchDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolverForPatch
+                );
+
+            patchDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        }
+
+        [Fact]
+        public void PatchDnsResolver_ResolverExistsEmptyRequest_ExpectSuccess()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolverForCreation = TestDataGenerator.GenerateDnsResolver(resourceGroupName: resourceGroupName);
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolverForCreation
+                );
+
+            var updatedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Update(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                parameters: dnsResolverForCreation
+                );
+
+            updatedDnsResolver.Should().NotBeNull();
```

Comment:
How do we know if this request didn't actually remove tags or something? We should verify the state of the resolver is as expected after call.

Reply:
BeSuccessfullyUpdatedWithTags

Reply:
BeSuccessfullyUpdatedWithTags